### ISSUE TITLE
Feature request - key handling

### ISSUE THREAD

**Issue by rg00dwin** [jpfxgood/ped#15]
escape key - exit w/out save - maybe an "are you sure?" guardrail w/ option to not be asked again.

request: ignore unassigned key combinations: various unassigned ctrl keys exit the editor with an error - for example - ctrl f - crtl g - ctrl w - ctrl o - all cause the editor to exit with:

```
(py383) roy@bigboy:~/anaconda3/envs/py383/bin$ ./ped test.txt 
byte doesn't fit in chtype
(py383) roy@bigboy:~/anaconda3/envs/py383/bin$ 
```

ped.log:
[ped.log](https://github.com/jpfxgood/ped/files/4880009/ped.log)

**Comment by jpfxgood** [comment]
First, this isn't reproduceable... none of those keys exit, or exit without warning in the case of esc... the only reason you're exiting is that there is an unhandled exception
The exception that is in the log doesn't happen except on your system... however you have it set up...
If you look at the code where it occurs it literally gets a character from curses and then pushes it back onto curses there is no change to the value being fetched
That type error indicates that for some reason your version of the curses wrapper is returning an incompatible type from getch such that you can't pass it to ungetch... that is a bug in their wrapper or somehow your configuration...
Are you building this anaconda distribution from source perhaps ? or using a development build ?

**Comment by rg00dwin** [comment]
Very interesting! OK - well good to know. 

I installed anaconda from the anaconda repo using these Digital Ocean instructions. 

https://www.digitalocean.com/community/tutorials/how-to-install-the-anaconda-python-distribution-on-ubuntu-20-04

All the packages in the py383 environment came with the environment when it was created with the exception of pygments, paramiko, and pytest which I added manually.

**Comment by jpfxgood** [comment]
Hah! I went back to my conda install and tried again and I was able to reprodce this crash and the original one... I also got it down to a tiny file that crashes ( if you remove the if ch == -1: continue part ) see below:

     import curses
      def main(stdscr):
          stdscr.nodelay(1)
          ch = -1
          while ch == -1:
              ch = stdscr.getch()
              print("The integer is %d"%ch,file=open("test.out","w"))
              if ch == -1:
                  continue
              print("The key pending is %d"%ch,file=open("test.out","w"))
              curses.ungetch(ch)
      
      if __name__ == '__main__':
          curses.wrapper(main)

So I think I can fix this in my code, but it still seems like a regression since ungetch never complained about getting passed a -1 in the past, logically I think it makes sense because there is nothing on the queue.

**Comment by rg00dwin** [comment]
Wow. That's pretty cool that you found it... This particular issue has been around in one form or another for years - at least if you look at the google entries about it.. I think curses has regressed in the past as well... 

In other news - I'm building a docker image (ubuntu 20.04LTS/python 3.83) to test out ped in - I want to try and get a nice tidy test environment safe from possible anomalies that could be added by anaconda.. Something simple.. clean.. reproducible.. etc..

**Comment by jpfxgood** [comment]
Yeah 3.8 has the problem their default which is like 3.7.3 no problem... also another wierdness on their 3.8 is that the backspace key is getting eaten... no idea why...

**Comment by jpfxgood** [comment]
I just pushed a fix for at least the crash bug to master, I'll have to write a bug about the backspace...

**Comment by rg00dwin** [comment]
yes - I noticed that Backspace was being ignored... I forgot to create an issue about it in all the excitement about testing in a docker container.. hah.

**Comment by jpfxgood** [comment]
fixed in 1.1.2
